compiler/ballerina-lang/src/main/java/io/ballerina/projects/BuildToolResolution.java

```diff
@@ -0,0 +1,315 @@
+        }
+        List<BuildTool> buildTools = new ArrayList<>();
+        Set<String> toolIds = new HashSet<>();
+        for (ToolContext toolContext: toolContextMap.values()) {
+            // Populate the tools needed to resolve
+            BuildToolId toolId = BuildToolId.from(toolContext.type().split("\\.")[0]);
+            if (toolIds.add(toolId.value())) {
+                buildTools.add(BuildTool.from(toolId, null, null, null));
+            }
+        }
+        ClassLoader toolClassLoader = this.getClass().getClassLoader();
+        ServiceLoader<CodeGeneratorTool> toolServiceLoader = ServiceLoader.load(
+                CodeGeneratorTool.class, toolClassLoader);
+
+        // Find tools which need resolution
+        List<BuildTool> resolutionRequiredTools = new ArrayList<>();
+        for (BuildTool tool : buildTools) {
+            Optional<CodeGeneratorTool> targetTool = BuildToolUtils.getTargetTool(tool.id().value(), toolServiceLoader);
+            if (targetTool.isEmpty()) {
+                resolutionRequiredTools.add(tool);
+            } else {
+                // If platform-provided, add to resolved tools with 0.0.0 version
+                tool.setVersion(PackageVersion.from(DEFAULT_VERSION));
+                tool.setOrg(PackageOrg.BALLERINA_ORG);
+                tool.setName(PackageName.from(PACKAGE_NAME_PREFIX + tool.id()));
+                resolvedTools.add(tool);
+            }
+        }
+        if (resolutionRequiredTools.isEmpty()) {
+            return;
+        }
+        try {
+            resolvedTools.addAll(resolveToolVersions(currentProject, resolutionRequiredTools));
+        } catch (CentralClientException e) {
+            throw new ProjectException("Failed to resolve tool dependencies: " + e.getMessage());
+        }
+    }
+
+    private List<BuildTool> resolveToolVersions(Project project, List<BuildTool> unresolvedTools)
+            throws CentralClientException {
+        PackageLockingMode packageLockingMode = getPackageLockingMode(project);
+        updateLockedToolDependencyVersions(unresolvedTools, project);
+        if (project.buildOptions().offlineBuild()) {
+            return getToolResolutionResponseOffline(unresolvedTools, packageLockingMode);
+        }
+        Set<ToolResolutionRequest> resolutionRequests = getToolResolutionRequests(unresolvedTools, packageLockingMode);
+        ToolResolutionCentralRequest toolResolutionRequest = createToolResolutionRequests(resolutionRequests);
+        return getToolResolutionResponse(toolResolutionRequest);
+    }
+
+    private PackageLockingMode getPackageLockingMode(Project project) {
+        boolean sticky = getSticky(project);
+
+        // new project
+        if (project.currentPackage().dependenciesToml().isEmpty()) {
+            if (sticky) {
+                return PackageLockingMode.MEDIUM;
+            }
+            return PackageLockingMode.SOFT;
+        }
+
+        // pre-built project
+        if (sticky) {
```

Comment:
Do we have a difference in the way we get sticky here compared to PackageResolution?

Reply:
Fixed with 38ea3e70e0481aaa222caab9223de33debdde6be